pkg/api/v1alpha1/cluster_defaults.go

```diff
@@ -45,8 +45,8 @@ func setRegistryMirrorConfigDefaults(clusterConfig *Cluster) error {
 }
 
 func setWorkerNodeGroupDefaults(cluster *Cluster) error {
-	if len(cluster.Spec.WorkerNodeGroupConfigurations) == 1 && cluster.Spec.WorkerNodeGroupConfigurations[0].Name == "" {
-		logger.V(1).Info("Worker node group name not specified. Defaulting name to md-0.")
+	if len(cluster.Spec.WorkerNodeGroupConfigurations) >= 1 && cluster.Spec.WorkerNodeGroupConfigurations[0].Name == "" {
```

Comment:
I think I would vote against this
I think if users are going to use the new feature, they should use the new syntax

Curious, what's the reason to propose this change?

Reply:
I proposed this change because if the user created the cluster with a version before 0.7.0, they would have to set the `name` of their first worker node group to `md-0` to have the node group supported during upgrade. If they set it to any other value, the original worker node group would be deleted during upgrade & it would create a new worker node group based on that new name.